### ISSUE TITLE
Persistent Download errors.

### ISSUE THREAD

**Issue by Mr-Kumar-Abhishek** [ronz94/Steadfast#7]
The downloader is giving download error warnings in every download. File downloaded is of zero bytes. Found after this commit:- ee9d5622409d0be2f0ad6f2f08ebb6fb2b6ce660

**Comment by ronz94** [comment]
Tried to create an object for the exception and then we need to stack trace with that object. Thus i think the error can be detected. ref: https://github.com/ronz94/Steadfast/commit/9f4d3798cd2b6ccb791be834b0c28eb16f2c38e2

**Comment by Mr-Kumar-Abhishek** [comment]
yes I have seen that , and reverted your last commit: 9f4d379. The bug is caused due to nullpointer exception:

```
java.lang.NullPointerException
    at javax.swing.Steadfast.actionPerformed(Steadfast.java:117)
    at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2018)
    at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2341)
    at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
    at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
    at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:252)
    at java.awt.Component.processMouseEvent(Component.java:6516)
    at javax.swing.JComponent.processMouseEvent(JComponent.java:3321)
    at java.awt.Component.processEvent(Component.java:6281)
    at java.awt.Container.processEvent(Container.java:2229)
    at java.awt.Component.dispatchEventImpl(Component.java:4872)
    at java.awt.Container.dispatchEventImpl(Container.java:2287)
    at java.awt.Component.dispatchEvent(Component.java:4698)
    at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4832)
    at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4492)
    at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4422)
    at java.awt.Container.dispatchEventImpl(Container.java:2273)
    at java.awt.Window.dispatchEventImpl(Window.java:2719)
    at java.awt.Component.dispatchEvent(Component.java:4698)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:740)
    at java.awt.EventQueue.access$300(EventQueue.java:103)
    at java.awt.EventQueue$3.run(EventQueue.java:699)
    at java.awt.EventQueue$3.run(EventQueue.java:697)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
    at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:87)
    at java.awt.EventQueue$4.run(EventQueue.java:713)
    at java.awt.EventQueue$4.run(EventQueue.java:711)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:710)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:242)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:161)
    at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:150)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:146)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:138)
    at java.awt.EventDispatchThread.run(EventDispatchThread.java:91)
```

Also you might want to check this ? :-  [see this gist](https://gist.github.com/Mr-Kumar-Abhishek/03e2f364e2db2125aa0d)

**Comment by ronz94** [comment]
detected the error and resolved the problem. https://github.com/ronz94/Steadfast/commit/96d3c90cca094e6917f81f9e20f511395da41552

**Comment by Mr-Kumar-Abhishek** [comment]
correction , resolved after commits 96d3c90 and d393c4f7448322173930529cae0e5cb3c911348f however 96d3c90 has not been neatly done. The file is being downloaded properly. And the file size is proper too. Also there are no persistant download errors now.
